diff -Naur shapelib-1.2.10/contrib/Makefile shapelib-1.2.10-mod/contrib/Makefile
--- shapelib-1.2.10/contrib/Makefile	2010-01-04 06:08:02.000000000 +0200
+++ shapelib-1.2.10-mod/contrib/Makefile	2010-02-16 10:18:48.000000000 +0200
@@ -32,7 +32,7 @@
 shpdata:	shpdata.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpdata.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpdata
 
-shpinfo:	shpinfo.c $(SHPOBJ)
+shpinfo:       shpinfo.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpinfo.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpinfo
 
 shpfix:		shpfix.c $(SHPOBJ)
@@ -64,3 +64,7 @@
 
 testproj: 
 	tests/shpproj.sh
+
+install: all
+	install dbfcat dbfinfo shpcat shpcentrd shpdata shpdxf shpfix shpinfo shpwkb shpproj \
+	$(bindir)
