diff -up shapelib-1.3.0b2/contrib/Makefile.makefile2 shapelib-1.3.0b2/contrib/Makefile
--- shapelib-1.3.0b2/contrib/Makefile.makefile2	2010-01-04 05:08:02.000000000 +0100
+++ shapelib-1.3.0b2/contrib/Makefile	2011-04-21 18:10:35.000000000 +0200
@@ -2,10 +2,7 @@
 #LINKOPT	=	/usr/local/lib/libdbmalloc.a
 #CFLAGS	=	-g
 
-# Endian: define either _LITTLE_ENDIAN or _BIG_ENDIAN
-ENDIAN	=	-D_LITTLE_ENDIAN
-
-CFLAGS	=	-g -I.. -I$(HOME)/bld/include -DPROJ4 $(ENDIAN) -DDEBUG -DDEBUG2 
+CFLAGS	=	-g -I.. -I$(HOME)/bld/include -DPROJ4 -DDEBUG -DDEBUG2 
 
 SHPOBJ	=	../shpopen.o ../dbfopen.o ../safileio.o
 
@@ -32,7 +29,7 @@ shpcentrd:	shpcentrd.c $(SHPGOBJ)
 shpdata:	shpdata.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpdata.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpdata
 
-shpinfo:	shpinfo.c $(SHPOBJ)
+shpinfo:	shpinfo.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpinfo.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpinfo
 
 shpfix:		shpfix.c $(SHPOBJ)
@@ -64,3 +61,7 @@ check:	testproj
 
 testproj: 
 	tests/shpproj.sh
+
+install: all
+	install dbfcat dbfinfo shpcat shpcentrd shpdata shpdxf shpfix shpinfo shpwkb shpproj \
+	$(bindir)
