diff -Naur shapelib-1.3.0b1/contrib/Makefile shapelib-1.3.0b1-mod/contrib/Makefile
--- shapelib-1.3.0b1/contrib/Makefile	2004-06-24 03:55:52.000000000 +0300
+++ shapelib-1.3.0b1-mod/contrib/Makefile	2010-01-07 09:42:05.000000000 +0200
@@ -1,13 +1,14 @@
+bindir = /usr/local/bin
 
 #LINKOPT	=	/usr/local/lib/libdbmalloc.a
 #CFLAGS	=	-g
 
 # Endian: define either _LITTLE_ENDIAN or _BIG_ENDIAN
-ENDIAN	=	-D_LITTLE_ENDIAN
+# ENDIAN	=	-D_LITTLE_ENDIAN
 
 CFLAGS	=	-g -I.. -DPROJ4 $(ENDIAN) -DDEBUG -DDEBUG2 
 
-SHPOBJ	=	../shpopen.o ../dbfopen.o 
+SHPOBJ	=	../shpopen.o ../dbfopen.o  ../safileio.o
 
 SHPGOBJ =	../shpopen.o ../dbfopen.o shpgeo.o
 
@@ -32,7 +33,7 @@
 shpdata:	shpdata.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpdata.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpdata
 
-shpinfo:	shpinfo.c $(SHPOBJ)
+shpinfo:	shpinfo.c $(SHPGOBJ)
 	$(CC) $(CFLAGS) shpinfo.c ${SHPOBJ} $(LINKOPT) $(GEOOBJ) -o shpinfo
 
 shpfix:		shpfix.c $(SHPOBJ)
@@ -64,3 +65,7 @@
 
 testproj: 
 	tests/shpproj.sh
+
+install: all
+	install dbfcat dbfinfo shpcat shpcentrd shpdata shpdxf shpfix shpinfo shpwkb shpproj \
+		$(bindir)
