--- shapelib-1.2.11-orig/contrib/Makefile	2006-01-07 12:34:14.000000000 -0800
+++ shapelib-1.2.11/contrib/Makefile	2006-03-04 20:11:01.000000000 -0800
@@ -1,9 +1,10 @@
+bindir = /usr/local/bin
 
 #LINKOPT	=	/usr/local/lib/libdbmalloc.a
 #CFLAGS	=	-g
 
 # Endian: define either _LITTLE_ENDIAN or _BIG_ENDIAN
-ENDIAN	=	-D_LITTLE_ENDIAN
+# ENDIAN	=	-D_LITTLE_ENDIAN
 
 CFLAGS	=	-g -I.. -DPROJ4 $(ENDIAN) -DDEBUG -DDEBUG2 
 
@@ -64,3 +65,7 @@
 
 testproj: 
 	tests/shpproj.sh
+
+install: all
+	install dbfcat dbfinfo shpcat shpcentrd shpdata shpdxf shpfix shpinfo shpwkb shpproj \
+		$(bindir)
